Improve description in readme and extract internals (#192)
yiisoft/yii-console: README.md

[![static analysis](https://github.com/yiisoft/yii-console/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/yii-console/actions?query=workflow%3A%22static+analysis%22)
[![type-coverage](https://shepherd.dev/github/yiisoft/yii-console/coverage.svg)](https://shepherd.dev/github/yiisoft/yii-console)

Yii Console package provides a console that could be added to an application. This console based on 
[Symfony Console](https://github.com/symfony/console), and added extra features:

- lazy command loader;

> - `SymfonyEventDispatcher` class that allow use any [PSR-14](http://www.php-fig.org/psr/psr-14/) compatible event 

Reviewer: ```suggestion
- `SymfonyEventDispatcher` class that allows to use any [PSR-14](http://www.php-fig.org/psr/psr-14/) compatible event 
```

  dispatcher with Symfony console;
- `ErrorListener` for logging console errors to any [PSR-3](https://www.php-fig.org/psr/psr-3/) compatible logger;

> - console command `serve` to runs PHP built-in web server;

Reviewer: ```suggestion
- console command `serve` that runs PHP built-in web server;
```

> - throw events `ApplicationStartup` and `ApplicationShutdown` in console application;

Reviewer: ```suggestion
- raises events `ApplicationStartup` and `ApplicationShutdown` in console application;
```

> - class `ExitCode` that contain constants for defining console command exit codes;

Reviewer: ```suggestion
- class `ExitCode` that contains constants for defining console command exit codes;
```

> - `ConsoleBufferedOutput` that wrap `ConsoleOutput` and buffers console output.  

Reviewer: ```suggestion
- `ConsoleBufferedOutput` that wraps `ConsoleOutput` and buffers console output.  
```
